New package: fastfetch-2.13.2

fvalasiad opened the pull request:
A solid neofetch replacement for void powerusers!

Closes #38293

#### Testing the changes
- I tested the changes in this PR: **briefly**

#### New package
- This new package conforms to the [package requirements](https://github.com/void-linux/void-packages/blob/master/CONTRIBUTING.md#package-requirements): **YES**

#### Local build testing
- I built this PR locally for my native architecture, (x86_64-glibc)
- I built this PR locally for these architectures (if supported. mark crossbuilds):
  - x86_64-musl
  - aarch64-glibc

Bnyro commented:
Reasoning: Neofetch is discontinued/archived, see https://news.itsfoss.com/neofetch-rip/

justinesmithies commented:
Fastfetch-2.11.5 is out now.

fvalasiad commented:
> Fastfetch-2.11.5 is out now.

I am aware, thanks!

They are releasing almost daily though and that would be too spammy for the void team, so if the package gets merged, I'll aim towards updating once a week or so. I will have to test the package before releasing each time so it's for the best regarding my own time too!

I will update the PR for now, but rest assured I will be updating it in the future without a need for a ping ( I hope!), don't think I gave up on it if I skipped a release or two!

AnInternetTroll reviewed (changes requested):
You could consider running the tests for this package as well

```diff
diff --git a/srcpkgs/fastfetch/template b/srcpkgs/fastfetch/template
index 5dd8fb588b3..e3fa3a183ca 100644
--- a/srcpkgs/fastfetch/template
+++ b/srcpkgs/fastfetch/template
@@ -3,6 +3,7 @@ pkgname=fastfetch
 version=2.11.5
 revision=1
 build_style=cmake
+configure_args="-DBUILD_TESTS=On"
 short_desc="Fast neofetch alternative written in C"
 maintainer="Fotios Valasiadis <fvalasiad@gmail.com>"
 license="MIT"
@@ -10,6 +11,11 @@ homepage="https://github.com/fastfetch-cli/fastfetch"
 distfiles="https://github.com/fastfetch-cli/${pkgname}/archive/refs/tags/${version}.tar.gz"
 checksum="83b7699d0aee3aa1683721fe4b82d667c88e97e257d48e9efe586b0e830f8a64"
 
+do_check() {
+	cd build
+	ctest ${makejobs}
+}
+
 post_install() {
 	vlicense LICENSE
 }
```

AnInternetTroll commented:
Tested on x86\_64-glibc and works as expected!

fvalasiad commented:
@AnInternetTroll Surely! Patch applied.
Also added you as a Co-author

I've tested on x86_64-glibc, x86_64-musl, aarch64-glibc.

hervyqa commented:
please add changelog. like this #38310

fvalasiad commented:
> please add conf_file and changelog. like this #38310

Adding the changelog certainly, but uncertain about `conf_files`.

If that's meant to point to fastfetch's config, then according to its manpage:

> Config Files
>        Fastfetch uses JSONC based format for configuration. Fastfetch doesn't
>        generate config file automatically; it should be generated manually by
>        --gen-config.  The config file will be saved in
>        ~/.config/fastfetch/config.jsonc by default.

If you are willing to provide an extra pointer to what requesting I'll try my best to deliver!

AnInternetTroll commented:
On Sun, May 12, 2024 at 10:32:43AM GMT, Fotios Valasiadis wrote:
>> please add conf_file and changelog. like this #38310
>
>Adding the changelog certainly, but uncertain about `conf_files`.

Seems like fastfetch has an undocumented /etc/fastfetch/config.conf file
that takes in CLI arguments. For example:

```
--logo windows
--color cyan
--kernel-format {1} {2}
--host-format {2}
```

What @hervyqa is asking is you add
`conf_files="/etc/fastfetch/config.conf"` under `configure_args`, how
the PR linked does it

https://github.com/void-linux/void-packages/pull/38310/files#diff-4db011d7f520a56f51c21a61c5db944e05bafe1cbe4cbaa570f4099d2715a5eeR7

AnInternetTroll commented:
Side note, a better alternative might be to only build the tests when
the tests will be ran, and not always build them

```bash
if [ "$XBPS_CHECK_PKGS" ]; then
	configure_args="-DBUILD_TESTS=ON"
fi
```

AnInternetTroll commented:
On Sun, May 12, 2024 at 01:18:16PM GMT, Fotios Valasiadis wrote:
>Sure! `./xbps-src pkg fastfetch` fails with that though complaining that:
>
>> => ERROR: fastfetch-2.11.5_1: '/etc/fastfetch/config.conf' configuration file not in PKGDESTDIR!
>> => ERROR: fastfetch-2.11.5_1: cannot continue with installation!
>
>Any ideas?

On second though, maybe it's not a good idea

1. It's deprecated
> We are deprecating flags based config files (will be removed in v3.0.0).
> We suggest you migrate to json based config files. One may use -c
> /path/to/config.conf --gen-config to migrate existing flag based config
> files.

https://github.com/fastfetch-cli/fastfetch/blob/dev/CHANGELOG.md#240

2. The package doesn't need a "default config", so in theory it won't be
    overwritten by xbps

But thanks for testing it out!

fvalasiad commented:
> On Sun, May 12, 2024 at 01:18:16PM GMT, Fotios Valasiadis wrote: Sure! `./xbps-src pkg fastfetch` fails with that though complaining that: > => ERROR: fastfetch-2.11.5_1: '/etc/fastfetch/config.conf' configuration file not in PKGDESTDIR! > => ERROR: fastfetch-2.11.5_1: cannot continue with installation! Any ideas?
> On second though, maybe it's not a good idea 1. It's deprecated
> We are deprecating flags based config files (will be removed in v3.0.0). We suggest you migrate to json based config files. One may use -c /path/to/config.conf --gen-config to migrate existing flag based config files.
> https://github.com/fastfetch-cli/fastfetch/blob/dev/CHANGELOG.md#240 2. The package doesn't need a "default config", so in theory it won't be overwritten by xbps But thanks for testing it out!

Gucci, removed `configure_args` from the top and added a conditional append, just in case we wanna define default configure flags in the future. If that's alright with you!

AnInternetTroll commented:
Absolutely, great work!

SpidFightFR commented:
hey, i've been using fastfetch as well for the past 5 months or so, with a personal template i work on, it never broke for me.

I personally approve the addition of fastfetch as a replacement for neofetch. :v:
